### PR TITLE
Added ProgressessStatistics support

### PR DESCRIPTION
**Pull request by umbreak**
- Added `timestamp_value` to projections table to store the timestamp of the event (the current timestamp stored is the timestamp when the row was created/updated)
- Moved `Lens` to `kernel` module
- Carry the timestamp of the event in `Message`
- the `ProjectionProgress` timestamp is the timestamp of the event, not the timestamp of when something was stored
- Centralized the indexing of views to DistributedData and the triggering of views coordinator start/stop under the same stream
- Renamed `ProjectStatisticsCollection` to `ProjectCountsCollection` and `ProjectsStatistics` to `ProjectsCounts`
- Other small refactoring and fixes

### CODE REVIEW

**Review by bogdanromanx — DISMISSED** [review]

File: delta/migration/cassandra-init/cassandra_1_5.ddl [hunk 1]
```diff
@@ -134,13 +134,14 @@ CREATE TABLE IF NOT EXISTS delta_1_5_snapshot.snapshots
...
-    warnings      bigint,
-    failed        bigint,
-    value         text,
+    offset              timeuuid,
+    timestamp           bigint,
+    processed           bigint,
+    discarded           bigint,
+    warnings            bigint,
+    failed              bigint,
+    value               text,
```
[the remark this answers is not in the document]
Comment (umbreak) [reply delta/migration/cassandra-init/cassandra_1_5.ddl:143]: ? it wasn't there so I didn't add it

[the remark this answers is not in the document]
Comment (bogdanromanx) [reply delta/migration/cassandra-init/cassandra_1_5.ddl:143]: @imsdu could you please comment on this?

[the remark this answers is not in the document]
Comment (imsdu) [reply delta/migration/cassandra-init/cassandra_1_5.ddl:143]: We can add it but this table is small so it is not as interesting as in the large event tables

**Review by wwajerowicz — APPROVED** [review]

File: delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/routes/ElasticSearchViewsRoutes.scala [hunk 2]
```diff
@@ -37,16 +39,19 @@ import monix.execution.Scheduler
...
+  * @param progresse the statistics of the progresses for the elasticsearch views
   */
 final class ElasticSearchViewsRoutes(
     identities: Identities,
     acls: Acls,
     projects: Projects,
     views: ElasticSearchViews,
-    viewsQuery: ElasticSearchViewsQuery
+    viewsQuery: ElasticSearchViewsQuery,
+    progresse: ProgressesStatistics
```
Comment (wwajerowicz) [inline delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/routes/ElasticSearchViewsRoutes.scala:50]: progresses

**Review by wwajerowicz — APPROVED** [review]

**Review by bogdanromanx — APPROVED** [review]

**Review by wwajerowicz — APPROVED** [review]

**Pull request merged**